common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -882,7 +882,10 @@ public UTF8String rpad(int len, UTF8String pad) {
    */
   public UTF8String lpad(int len, UTF8String pad) {
     int spaces = len - this.numChars(); // number of char need to pad
-    if (spaces <= 0 || pad.numBytes() == 0) {
+    if (len > 0 && pad.numBytes() == 0) {
+      // no padding at all, return the null
+      return null;
+    } else if (spaces <= 0) {
```

Comment:
I don't think it's correct to fix it here since the issue is specific to `lpad` expression. Can you check `lpad` in other DBMSes? I think we shouldn't just blindly follow Hive

Reply:
ok and please check this PR also https://github.com/apache/spark/pull/26477

Reply:
I think it's fine not to fix for now.

Reply:
@HyukjinKwon, but later we need fix ?

Reply:
Current implementation is at least reasonable. If we should do, we could add a configuration to control the behaviour but to me it doesn't look worthy.

Reply:
But as per rpad/lpad definition 'In case of empty pad string, the return value is null.'  If we are not going to handle in 3.0 we can move to handle this in next version